src/crate/client/http.py

```diff
@@ -86,25 +86,31 @@ def super_len(o):
     return None
 
 
-class CrateJsonEncoder(json.JSONEncoder):
-    epoch_aware = datetime(1970, 1, 1, tzinfo=timezone.utc)
-    epoch_naive = datetime(1970, 1, 1)
-
-    def default(self, o):
-        if isinstance(o, (Decimal, UUID)):
-            return str(o)
-        if isinstance(o, datetime):
-            if o.tzinfo is not None:
-                delta = o - self.epoch_aware
-            else:
-                delta = o - self.epoch_naive
-            return int(
-                delta.microseconds / 1000.0
-                + (delta.seconds + delta.days * 24 * 3600) * 1000.0
-            )
-        if isinstance(o, date):
-            return calendar.timegm(o.timetuple()) * 1000
-        return json.JSONEncoder.default(self, o)
+epoch_aware = datetime(1970, 1, 1, tzinfo=timezone.utc)
+epoch_naive = datetime(1970, 1, 1)
+
+
+def cratedb_json_encoder(obj):
+    """
+    Encoder function for orjson.
+
+    https://github.com/ijl/orjson#default
+    https://github.com/ijl/orjson#opt_passthrough_datetime
+    """
```

Comment:
Like the [opt_passthrough_dataclass](https://github.com/ijl/orjson?tab=readme-ov-file#opt_passthrough_dataclass) option conveys it:

> Passthrough `dataclasses.dataclass` instances to `default`. This allows customizing their output but is much slower.

... I think the same holds true also for the [opt_passthrough_datetime](https://github.com/ijl/orjson?tab=readme-ov-file#opt_passthrough_datetime) option? JFYI.

Reply:
According to the [docs](https://github.com/ijl/orjson?tab=readme-ov-file#datetime), `orjson` serializes `DateTime` objects to a String format CrateDB understands. I wonder if we need this at all?

Reply:
I've added this admonition to the OP.

Reply:
Same for the `data` type, the string format `"1900-01-02"` is understood by CrateDB (see https://github.com/crate/crate/blob/master/server/src/test/java/io/crate/types/DateTypeTest.java#L45).

Also [UUID](https://github.com/ijl/orjson?tab=readme-ov-file#uuid) is supported and correctly serialized to a valid string. 

So seems like we only need any custom encoder for the `DECIMAL` type?

Reply:
> So seems like we only need any custom encoder for the `DECIMAL` type?

idk. I had created an issue about it over here the other day, if I am getting it right that `DECIMAL` ==  `NUMERIC`?

- https://github.com/crate/crate-python/issues/652

Reply:
> According to the [docs](https://github.com/ijl/orjson?tab=readme-ov-file#datetime), `orjson` serializes `DateTime` objects to a String format CrateDB understands. I wonder if we need this at all?

I actually was wondering about it as well. However, I started bringing this in by keeping it as a fact that we'd need a custom decoder. Thanks for bringing it up: Let's try to swap it out and check what CI will tell us.

Reply:
> if I am getting it right that DECIMAL == NUMERIC

Yes.
I've just [added a comment](https://github.com/crate/crate-python/issues/652#issuecomment-2595082040) to the mentioned issue as it's not really clear to me what is missing to support `decimal` values, it should already work.

Reply:
With two more adjustments to test cases, the test suite succeeds also without `OPT_PASSTHROUGH_DATETIME`. If you will confirm this is a good idea, the positive consequence is to trim the new `cratedb_json_encoder` function significantly?

Reply:
When adding 03559a9028, the test suite [starts failing](https://github.com/crate/crate-python/actions/runs/12806754738/job/35705886534#step:5:177).

Expectations are that unixtime values are exchanged, while it suddenly became a native Python datetime object.                                                                                                                                                                                                                                                                                                               
```
 AssertionError: datetime.datetime(2023, 6, 26, 9, 24, 0, [62 chars]00))) != 1687764240123
```

Reply:
Because crate-2.0.0.dev1 works [so excellently](https://github.com/crate/crate-python/pull/691#discussion_r1918115729), I'd tend to skip this within the current iteration on this patch, and possibly add it as an improvement later.

Reply:
Because @seut says it works like [this](https://github.com/crate/crate-python/pull/691#pullrequestreview-2555959462), we are eating the free lunch now, and are able to remove quite an amount of code.

Thanks, @seut!